core/cap-0034.md

```diff
@@ -0,0 +1,352 @@
+allow transactions dependent on `A` (which the smart contract chains after `A`)
+to be executed, and if any of them succeeds, then they will perform operations
+which the smart contract intended to be performed only if the operations of `A`
+had been performed.  This race is probably uncommon, because smart contracts
+tend to set expiration times much farther out than SDKs (on the order of
+anywhere from hours to years), but potentially arbitrarily bad for a smart
+contract that did hit it.
+
+Additionally, externalizing transaction sets that contain significant numbers
+of transactions that are doomed to fail with `txTOO_LATE` during ledger close
+might mean that we're choosing sets that we would have considered worse than
+other available candidates if we had taken into account that those transactions
+were doomed.
+
+### Goals Alignment
+
+This CAP is aligned with the following Stellar Network Goals:
+
+- The Stellar Network should be secure and reliable.  Though the
+events that could lead to a smart contract executing inconsistently because of
+the problem that this CAP intends to address might be very rare, they might
+also be of a kind of which even a single event could significantly damage trust.
+A "reliable" network must try to avoid _any_ behavior which could appear
+sufficiently unpredictable to a client that they would be unlikely to consider
+it and could obtain inconsistent results when it happened.
+
+- The Stellar Network should run at scale and at low cost to all users.
+Keeping transactions which are guaranteed to fail later in
+externalized transaction sets is wasteful.
+
+It also aligns with the following Stellar Network Value:
+
+- The protocol should bias toward simplicity.  This CAP simplifies the
+ballot protocol by removing the "combination" of `closeTime`s and simply
+selecting the one associated with the selected transaction set.
+
+## Abstract
+
+We change the (deterministic) mechanism by which the ballot protocol computes
+the `StellarValue` to be externalized from producing a value comprising the
+best transaction set (selected by a deterministic heuristic which favors larger
+sets) of any candidate, the maximum `closeTime` of any candidate, and a set of
+"maximal" ledger `upgrades` from all candidates to one which takes the
+`closeTime` from the same candidate `StellarValue` as the best transaction set
+(the heuristic which decides which transaction set is best does not change;
+nor does the method of generating `upgrades`, which remains a combination).
+
+In particular, this affects how soon the core notices when transactions expire.
+
+## Specification
+
+There are no changes to any XDR in this CAP.  The treatment of the `closeTime`
+in the `StellarValue` XDR in some code paths changes, as does the use of
+`SIGNED` `StellarValue`s in some places where `BASIC` ones are currently used,
+but the XDR itself does not change.
+
+To date, the ballot protocol has generated a value to externalize by combining
```

Comment:
You can just remove this paragraph entirely from this section. This is the specification: I want to know exactly how it will work but everything else is irrelevant.

Reply:
Done.